packages/axios-extension/src/abap/lrep-service.ts

```diff
@@ -0,0 +1,163 @@
+import type { Service } from '../base/service-provider';
+import type { AxiosResponse } from 'axios';
+import { Axios } from 'axios';
+import { LogLevel } from '@sap-ux/logger';
+import type { Logger } from '@sap-ux/logger';
+import { readFileSync } from 'fs';
+
+/**
+ * Type representing a namespace. It is either a string or an object containing an id (variant id) and a reference (base application id).
+ */
+export type Namespace =
+    | string
+    | {
+          /**
+           * variant id
+           */
+          id: string;
+
+          /**
+           * base application id
+           */
+          reference: string;
+      };
+
+/**
+ * Required configuration to deploy an adaptation project.
+ */
+export interface AdaptationConfig {
+    /**
+     * Namespace either as string or object
+     */
+    namespace: Namespace;
+
+    /**
+     * Optional ABAP package name
+     */
+    package?: string;
+
+    /**
+     * Optional transport request
+     */
+    transport?: string;
+}
+
+/**
+ * Structure of the result message.
+ */
+export interface Message {
+    severity: 'Success' | 'Warning' | 'Error';
+    text: string;
+    details?: string[];
+    id: string;
+    variables?: string[];
+}
+
+/**
+ * Technically supported layers, however, in practice only `CUSTOMER_BASE` is used
+ */
+type Layer = 'VENDOR' | 'CUSTOMER_BASE';
+
+/**
+ * Returns the namespace as string.
+ *
+ * @param namespace either as string or as object
+ * @returns serialized namespace
+ */
+function getNamespaceAsString(namespace: Namespace): string {
+    return namespace['id'] ? `apps/${namespace['reference']}/appVariants/${namespace['id']}/` : (namespace as string);
```

Comment:
Does this work better with an explicit type check? If `string` ever has an `id` property this will silently break

Reply:
Done with https://github.com/SAP/open-ux-tools/pull/469/commits/3de6c2502ddeaa17de391f8a733d3f542bd116be.